### ISSUE TITLE
FileMigrationHelper doesn't create live records for folders

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-assets#373]
Folders aren't versioned themselves, and don't distinguish between draft and live stages. Only the files within them exhibit that behaviour. But: When using Versioned-specific ORM queries, those are sensitive to draft/live stage even on folders. Which is why the asset abstraction writes them to *both* stages by default. But: The file migration script does *not* do this for existing folders.

Example:

```
Versioned::set_stage('Draft');
Folder::find('my-legacy-folder')->exists(); // true
Versioned::set_stage('Live');
Folder::find('my-legacy-folder')->exists(); // false

Versioned::set_stage('Draft');
Folder::find('my-ss4-created-folder')->exists(); // true
Versioned::set_stage('Live');
Folder::find('my-ss4-created-folder')->exists(); // true
```

This can have severe unintended consequences when folder existence is used to determine an upload location (with a fallback).

### ACs

- [x] The file migration process during a CMS 4 upgrade ensures that the most recent data for every migrated Folder is published to the Folder_Live table
- [x] This has no / negligible impact on performance of the migration flow
- [x] Provides clear signal that this operation has occurred

## Related

 * [Multiple "Uploads" folders are being created](https://github.com/silverstripe/silverstripe-assets/issues/330)

**Comment by chillu** [comment]
I've created a related issue about duplicate folders: https://github.com/silverstripe/silverstripe-assets/issues/374

**Comment by dnsl48** [comment]
> the asset abstraction writes [them] to both stages by default.

Happens in [onAfterWrite](https://github.com/silverstripe/silverstripe-assets/blob/1.5.1/src/Folder.php#L241)

**Comment by chillu** [comment]
The ACs are a bit ambiguous on this, so I'll spell it out here: Devs need to be able to run the task "standalone" (without other tasks) for sites which have already been migrated to 4.x, but should *not* need to worry about this on any future migrations to 4.x. In that case, the task should just run alongside the default execution of `dev/tasks/MigrateFileTask`. I think the easiest way to achieve this is to fold it into `MigrateFileTask.php`, and allow running it via `dev/tasks/MigrateFileTask task=fixfolders`

**Comment by dnsl48** [comment]
This has been addressed in [4.4.6](https://docs.silverstripe.org/en/4/changelogs/4.4.6/) and [4.5.2](https://docs.silverstripe.org/en/4/changelogs/4.5.2/) releases.

The issue is not actually related to https://github.com/silverstripe/silverstripe-assets/issues/330, since that issue describes creation of multiple duplicates.
In this issue we are talking about a single "broken" folder with a draft-only record, and its duplicate with both draft-and-live records. Creating more then one "broken" folder is not possible through merely migrating it from SS3, and creation of new folders in SS4 always generates Live versions.
